### ISSUE TITLE
Links in "Select branch/tag" menu always point to upstream

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#30879]
Type: <b>Bug</b>

Steps to reproduce:
1. you need a git repo with origin (your fork) and upstream (the project you forked). now you have: origin/main and upstream/main
2. on the VSCode statusbar click "Select Branch/Tag...", so that the menu with "Select branch or tag for checkout..." opens
3. This menu contains additional links for "origin/main" and "upstream/main" entries:
the first icon is Github - "Open on Github"
the second icon is World - "Checkout on vscode.dev"

Issue:
When you click the "Open on Github" icon for the `origin/main` repo
it will open the repository website for `upstream/main`.
This is also true for all other branches.
In other words: all links point to the webpage of the parent repo `upstream/main`. You can not open webpages for branches like `origin/my-fix`. This results in 404, because it tries to open the webpage for `upstream` with branch `my-fix`, which exists only on the child repo.

Expected:
When clicking "Open on Github" for `origin/main` the webpage for the origin repo should open.
When clicking "Open on Github" for `upstream/main` the webpage for the upstream repo should open.

VS Code version: Code 1.82.2 (abd2f3db4bdb28f9e95536dfa84d8479f1eb312d, 2023-09-14T05:55:25.390Z)
OS version: Windows_NT x64 10.0.19045
Modes:
